### PR TITLE
Upgrade apex-parser to new version 2.16.

### PR DESCRIPTION
**Pull request by aaronhurst-google**
AST changes supporting new grammar:
* Eliminate non-existent modulo and modulo-assign operators
* Support for and while loops with semicolon bodies
* Placeholder exception for void.Class primaries until semantics are defined

Update next Summit version to 2.0.0-SNAPSHOT due to non-backward-compatible changes.

### CODE REVIEW

File: src/main/java/com/google/summit/ast/expression/BinaryExpression.kt [hunk 1]
```diff
@@ -47,7 +47,6 @@ class BinaryExpression(
     SUBTRACTION,
     MULTIPLICATION,
     DIVISION,
-    MODULO,
```
Comment (zorzella) [inline src/main/java/com/google/summit/ast/expression/BinaryExpression.kt:50]: just curious why was this here to begin with?

**Review by zorzella — APPROVED** [review]

File: src/main/java/com/google/summit/ast/expression/BinaryExpression.kt (hunk 1 above)
Comment (aaronhurst-google) [reply src/main/java/com/google/summit/ast/expression/BinaryExpression.kt:50]: I more-or-less used the grammar as the language spec. The grammar had a modulo operator, even though the language does not. This error was rectified in the latest grammar update. (I was curious enough to go look at the online Apex language reference and confirmed.)
